### PR TITLE
Adds syntax highlighting

### PR DESCRIPTION
**Pull request by emersion**
See #26, #13.

### CODE REVIEW

**Comment by voldyman** [comment]
can you rebase this branch on voldyman:discount-2.1.8 ?

**Comment by emersion** [comment]
Done!

**Comment by emersion** [comment]
Done!

File: src/Widgets/Window.vala [hunk 1]
```diff
@@ -270,20 +274,113 @@ public class Window : Gtk.Window {
         return false;
     }
 
+    private string get_data_file_uri (string filename) {
+        File file = File.new_for_path ("../data/assets/"+filename);
+        if (file.query_exists ()) {
+            return file.get_uri ();
+        }
+
+        var data_dirs = Environment.get_system_data_dirs ();
```
Comment (voldyman) [inline src/Widgets/Window.vala]: This should use Constants.PKGDATADIR to find the file

File: src/Widgets/Window.vala [hunk 2]
```diff
@@ -270,20 +274,113 @@ public class Window : Gtk.Window {
...
+        string processed_mk;
+        process_frontmatter (raw_mk, out processed_mk);
+
+        var mkd = new Markdown.Document (processed_mk.data);
         mkd.compile ();
 
         string result;
         mkd.get_document (out result);
 
         string html = "<html><head>";
```
Comment (voldyman) [inline src/Widgets/Window.vala]: do we have to generate the html of every refresh? too many file io's

Comment (emersion) [inline src/Widgets/Window.vala]: Yes, I was planning to cache files contents in variables, that would be a lot better.

**Comment by voldyman** [comment]
ok, we need to merge this branch which requires  that you rebase it on the current master (which i am not going to touch for a few days)

i thought i had merged this sometime ago, thanks for the contribution :heart:

**Comment by emersion** [comment]
Just merged with upstream and added assets caching ;-)

**Comment by voldyman** [comment]
:+1:

**Pull request merged**